### PR TITLE
Add support for configuring nickname and color

### LINKED ISSUE #142: Possibility for players to change color in game
Currently, players are having random colors, maybe it would be great to add any UI for players to be able to change the color.

### PR DESCRIPTION
**Pull request by PhantomGamers**
Color is currently configured with separate RGB text boxes.

I was going to enable the slider but can't because of #285 

Adding Float3 support to the menu with an actual color picker would be ideal but this works for now.

Closes #142 #143

### CODE REVIEW

File: NebulaModel/MultiplayerOptions.cs [hunk 1]
```diff
@@ -7,6 +7,21 @@ namespace NebulaModel
     [System.Serializable]
     public class MultiplayerOptions : ICloneable
     {
+        [DisplayName("Nickname")]
+        public string Nickname { get; set; } = "Player";
```
Comment (hubastard) [inline NebulaModel/MultiplayerOptions.cs:11]: We could use `AccountData.me.userName` as default value

File: NebulaModel/MultiplayerOptions.cs [hunk 2]
```diff
@@ -7,6 +7,21 @@ namespace NebulaModel
...
+        [UIRange(0f, 1f)]
+        public float ColorR { get; set; } = 1f;
+
+        [DisplayName("Color Green")]
+        [UIRange(0f, 1f)]
+        public float ColorG { get; set; } = 0.6846404f;
+
+        [DisplayName("Color Blue")]
+        [UIRange(0f, 1f)]
+        public float ColorB { get; set; } = 0.24313718f;
```
Comment (hubastard) [inline NebulaModel/MultiplayerOptions.cs:23]: We should use the name `Mecha Color Red` to be more explicit

File: NebulaModel/MultiplayerOptions.cs (hunk 1 above)
Comment (PhantomGamers) [reply NebulaModel/MultiplayerOptions.cs:11]: I wanted to do this but the config generates before AccountData.me.userName is populated

File: NebulaModel/MultiplayerOptions.cs (hunk 2 above)
Comment (PhantomGamers) [reply NebulaModel/MultiplayerOptions.cs:23]: > We should use the name `Mecha Color Red` to be more explicit

Should I reflect this in the variable name as well?

File: NebulaModel/MultiplayerOptions.cs (hunk 1 above)
Comment (Shad0wlife) [reply NebulaModel/MultiplayerOptions.cs:11]: Do you know when that value gets populated and then auto-set this value to it? Or is that not viable (performance, or the population is too late)?

File: NebulaModel/MultiplayerOptions.cs (hunk 2 above)
Comment (hubastard) [reply NebulaModel/MultiplayerOptions.cs]: yes please

Comment (PhantomGamers) [reply NebulaModel/MultiplayerOptions.cs]: > yes please

Done in 58edede

File: NebulaModel/MultiplayerOptions.cs (hunk 1 above)
Comment (PhantomGamers) [reply NebulaModel/MultiplayerOptions.cs:11]: I could keep the default null and make a postfix patch for PARTNER.Login() and do Config.Options.Nickname = AccountData.me.userName ?? "Player"; there I suppose.

Would this be desireable?

Some games, though usually older ones, with official multiplayer support default to just calling the player "Player" so I figured this was just the cleanest approach but it's up to you folks.

**Review by Therzok — APPROVED** [review]

**Pull request closed without merging**